src/bcccp/carpark/Carpark.java

```diff
@@ -1,150 +1,170 @@
-		// TODO Auto-generated method stub
-		
-	}
 
+    @Override
+    public void register(ICarparkObserver observer) {
 
+        observers.add(observer);
 
-	@Override
-	public String getName() {
-		// TODO Auto-generated method stub
-		return null;
-	}
+    }
 
 
+    @Override
+    public void deregister(ICarparkObserver observer) {
 
-	@Override
-	public boolean isFull() {
-		// TODO Auto-generated method stub
-		return false;
-	}
+        observers.remove(observer);
+        numberOfCarsParked--;
 
+    }
 
 
-	@Override
-	public IAdhocTicket issueAdhocTicket() {
-		// TODO Auto-generated method stub
-		return null;
-	}
+    @Override
+    public String getName() {
 
+        return carparkId;
+    }
 
 
-	@Override
-	public void recordAdhocTicketEntry() {
-		// TODO Auto-generated method stub
-		
-	}
+    @Override
+    public boolean isFull() {
 
+        return (numberOfCarsParked >= capacity) ? true : false ;
+    }
 
 
-	@Override
-	public IAdhocTicket getAdhocTicket(String barcode) {
-		// TODO Auto-generated method stub
-		return null;
-	}
+    @Override
+    public IAdhocTicket issueAdhocTicket() {
 
+        return adhocTicketDAO.createTicket(carparkId);
+    }
 
 
-	@Override
-	public float calculateAddHocTicketCharge(long entryDateTime) {
-		// TODO Auto-generated method stub
-		return 0;
-	}
+    @Override
+    public void recordAdhocTicketEntry() {
 
+        numberOfCarsParked++;
 
+    }
 
-	@Override
-	public void recordAdhocTicketExit() {
-		// TODO Auto-generated method stub
-		
-	}
 
+    @Override
+    public IAdhocTicket getAdhocTicket(String barcode) {
 
+        return adhocTicketDAO.findTicketByBarcode(barcode);
+    }
 
-	@Override
-	public void registerSeasonTicket(ISeasonTicket seasonTicket) {
-		// TODO Auto-generated method stub
-		
-	}
 
+    @Override
+    public float calculateAddHocTicketCharge(long entryDateTime) {
+        // Calculation: get current Date and Time, subtract entryDateTime, multiply result by
+        // the $ charge rate and return charge. Assumption that rates are a fixed rate of $5.00 per hour.
+        // Convert time to hours by dividing by 60,000. Assumption this is short-stay tariff car park.
 
+        float rates = 5.0f;
 
-	@Override
-	public void deregisterSeasonTicket(ISeasonTicket seasonTicket) {
-		// TODO Auto-generated method stub
-		
-	}
+        Date dateTime = new Date();
 
+        float chargeAmount = (dateTime.getTime() - entryDateTime) * rates / 60000;
 
+        return chargeAmount;
+    }
 
-	@Override
-	public boolean isSeasonTicketValid(String ticketId) {
-		// TODO Auto-generated method stub
-		return false;
-	}
 
+    @Override
+    public void recordAdhocTicketExit() {
 
+        numberOfCarsParked--;
 
-	@Override
-	public boolean isSeasonTicketInUse(String ticketId) {
-		// TODO Auto-generated method stub
-		return false;
-	}
+    }
 
 
+    @Override
+    public void registerSeasonTicket(ISeasonTicket seasonTicket) {
```

Comment:
JAVADOC: Please add javadoc as per Google Java Style guide.
At the minimum, Javadoc is present for every public class, and every public or protected member of such a class.